### ISSUE TITLE
Feature Request: evaluate both train_data and test_data using tf.contrib.learn.Experiment?

### ISSUE THREAD

**Issue by mufusu** [tensorflow/tensorflow#19451]
### Describe the problem
feature request：
i'm using the api "tf.contrib.learn.Experiment" for distributed training, but i don't know the method for evaluating the train_data and test_data at the same step, i want to check the model is overfit or not, so i want to print both train_auc and test_auc after saving model each time.

now, i can only see the performance on the test data for every saved model. is there any methods to print both train and test evaluation? Thanks a lot!

### Source code / logs
here is  code:
`def train(self):
        if self.job_name == "worker":
            # get data
            self.init_worker_input_split()
            self.init_worker_test_input_split()

        # data input function
        train_input = lambda: dataset_v1.create_input_fun(self.file_names,
                                                          FLAGS.batch_size,
                                                          repeat_count=FLAGS.train_epochs)
        test_input = lambda: dataset_v1.create_input_fun(self.test_file_names,
                                                         FLAGS.batch_size,
                                                         perform_shuffle=False)
        # estimator define
        model = self.build_estimator(FLAGS.log_dir, FLAGS.model_type)

        distribute_exp = tf.contrib.learn.Experiment(estimator=model,
                                                     train_input_fn=train_input,
                                                     eval_input_fn=test_input,
                                                     min_eval_frequency=100
                                                     )

        if self.job_name == "ps":
            # Starts a TensorFlow server and joins the serving thread
            distribute_exp.run_std_server()
        elif self.job_name == "worker":
            distribute_exp.train_and_evaluate()

        if self.is_chief:
            # Evaluate on the evaluation data.
            distribute_exp.evaluate(name='test_performance')

        tf.logging.info("Optimization Finished!")`

here is the logs:
INFO:tensorflow:Validation (step 7014908): loss = 15.8598, accuracy_baseline = 0.925938, global_step = 7014348, auc = 0.683994, prediction/mean = 0.0714387, label/mean = 0.0740625, average_loss = 0.24781, auc_precision_recall = 0.167725, accuracy = 0.927031
INFO:tensorflow:global_step/sec: 98.1175
INFO:tensorflow:global_step/sec: 124.708
INFO:tensorflow:global_step/sec: 121.122
INFO:tensorflow:loss = 11.3313, step = 7016362 (21.941 sec)
INFO:tensorflow:global_step/sec: 122.56
INFO:tensorflow:global_step/sec: 114.473
INFO:tensorflow:global_step/sec: 122.207
INFO:tensorflow:global_step/sec: 123.767
INFO:tensorflow:global_step/sec: 125.336
INFO:tensorflow:loss = 14.5477, step = 7016886 (4.307 sec)
INFO:tensorflow:global_step/sec: 120.105
INFO:tensorflow:global_step/sec: 110.698
INFO:tensorflow:global_step/sec: 124.799
INFO:tensorflow:global_step/sec: 121.776
INFO:tensorflow:global_step/sec: 114.484
INFO:tensorflow:loss = 16.5542, step = 7017438 (4.837 sec)
INFO:tensorflow:global_step/sec: 125.595
INFO:tensorflow:global_step/sec: 122.375
INFO:tensorflow:global_step/sec: 122.392
INFO:tensorflow:global_step/sec: 124.201
INFO:tensorflow:global_step/sec: 122.038
INFO:tensorflow:loss = 17.7121, step = 7017930 (3.817 sec)
INFO:tensorflow:global_step/sec: 102.7
INFO:tensorflow:global_step/sec: 121.585
INFO:tensorflow:global_step/sec: 119.404
INFO:tensorflow:global_step/sec: 121.501
INFO:tensorflow:global_step/sec: 119.815
INFO:tensorflow:loss = 16.0022, step = 7018577 (5.630 sec)
INFO:tensorflow:global_step/sec: 118.615
INFO:tensorflow:global_step/sec: 122.272
INFO:tensorflow:global_step/sec: 121.206
INFO:tensorflow:global_step/sec: 123.087
INFO:tensorflow:global_step/sec: 120.079
INFO:tensorflow:loss = 20.2255, step = 7019070 (4.049 sec)
INFO:tensorflow:global_step/sec: 118.552
INFO:tensorflow:global_step/sec: 121.404
INFO:tensorflow:Saving checkpoints for 7019347 into viewfs://hadoop-meituan/user/hadoop-generalshop/caiqi.sun/spark/dl/demo/output/wide_deep_all_82/model.ckpt.
INFO:tensorflow:global_step/sec: 102.859
INFO:tensorflow:Starting evaluation at 2018-05-22-03:56:24
INFO:tensorflow:Restoring parameters from viewfs://hadoop-meituan/user/hadoop-generalshop/caiqi.sun/spark/dl/demo/output/wide_deep_all_82/model.ckpt-7019347
INFO:tensorflow:Evaluation [10/100]
INFO:tensorflow:Evaluation [20/100]
INFO:tensorflow:Evaluation [30/100]
INFO:tensorflow:Evaluation [40/100]
INFO:tensorflow:Evaluation [50/100]
INFO:tensorflow:Evaluation [60/100]
INFO:tensorflow:Evaluation [70/100]
INFO:tensorflow:Evaluation [80/100]
INFO:tensorflow:Evaluation [90/100]
INFO:tensorflow:Evaluation [100/100]
INFO:tensorflow:Finished evaluation at 2018-05-22-03:56:31
INFO:tensorflow:Saving dict for global step 7019354: accuracy = 0.927031, accuracy_baseline = 0.925938, auc = 0.683203, auc_precision_recall = 0.170481, average_loss = 0.247913, global_step = 7019354, label/mean = 0.0740625, loss = 15.8664, prediction/mean = 0.0705214
INFO:tensorflow:Validation (step 7019922): loss = 15.8664, accuracy_baseline = 0.925938, global_step = 7019354, auc = 0.683203, prediction/mean = 0.0705214, label/mean = 0.0740625, average_loss = 0.247913, auc_precision_recall = 0.170481, accuracy = 0.927031

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 19 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 34 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
